cloud/aws/bin/pgadmin.py

```diff
@@ -0,0 +1,107 @@
+import os
+import sys
+import time
+import re
+import urllib.error
+import urllib.request
+
+from cloud.aws.templates.aws_oidc.bin import resources
+from cloud.aws.templates.aws_oidc.bin.aws_cli import AwsCli
+from cloud.shared.bin.lib import terraform
+
+
+def run(config):
+    os.environ["TF_VAR_pgadmin_cidr_allowlist"] = get_cidr_list()
+    os.environ["TF_VAR_pgadmin"] = "true"
+
+    if not terraform.perform_apply(config):
+        sys.stderr.write("Terraform deployment failed.")
+        raise ValueError("Terraform deployment failed.")
+
+    pgurl = f"{config.get_base_url()}:4433"
+    print(
+        "\npgadmin terraform deployment finished. Waiting for pgadmin service (some request failures are expected):"
+    )
+    wait_for_pgadmin_response(pgurl)
+
+    print(f"\npgadmin service is available. URL: {pgurl}")
+    print_secrets(config)
+
+    input(
+        "\nWARNING: it is strongly recommended to clean up pgadmin resources once they are no-longer needed.\n"
+        "Run 'bin/deploy --tag=latest' to manually clean up pgadmin resources.\n\n"
+        "Waiting to clean up pgadmin resources.  Press enter to trigger cleanup...\n"
+    )
+
+    os.unsetenv("TF_VAR_pgadmin_cidr_allowlist")
+    os.unsetenv("TF_VAR_pgadmin")
+    if not terraform.perform_apply(config):
+        sys.stderr.write("Terraform deployment failed.")
+        raise ValueError("Terraform deployment failed.")
+
+
+def get_cidr_list() -> str:
+    print()
+
+    while True:
+        got = input(
+            "\nREQUIRED: input a comma-separated list of IPv4 CIDR blocks that should have access to the pgadmin service:\n"
+            "To allow a single IP address, use a subnet mask of 32. For example: '192.0.0.1/32'\n"
```

Comment:
That need more instructions. Personally I have no idea what to put here. Like I can run `ifconfig` and try to find one of the ips there. But generally that's not correct if I'm sitting behind a NAT and my public IP is different, right? Or provide an option here like "if you ok allowing all ips use 0.0.0.0/..." or whatever is the allow-all address.

Reply:
Maybe there is a way to make a call to some public service that gives you your public ip? Then you can just propose using it.

Reply:
Done.